rust/worker/src/index/hnsw.rs

```diff
@@ -202,7 +202,7 @@ impl Index<HnswIndexConfig> for HnswIndex {
     }
 
     fn add(&self, id: usize, vector: &[f32]) {
-        unsafe { add_item(self.ffi_ptr, vector.as_ptr(), id, false) }
+        unsafe { add_item(self.ffi_ptr, vector.as_ptr(), id, true) }
```

Comment:
For my learning, what is the meaning of this `replace_deleted` param? Does this mean that hnsw keeps deleted items around?

Reply:
it means hnsw will reuse deleted space in its allocated buffer